### PR TITLE
[#1022] Change ingress hostnames and deploy traefik ingress controller

### PR DESCRIPTION
**Pull request by ljupcovangelski**

### CODE REVIEW

File: infrastructure/helm-chart/templates/ingress.yaml [hunk 1]
```diff
@@ -297,7 +288,7 @@ spec:
                 name: sources-chatplugin
                 port:
                   number: 80
-          - path: /
+          - path: /ui/chatplugin
```
Comment (chrismatix) [inline infrastructure/helm-chart/templates/ingress.yaml:291]: I am assuming that there would be a conflict with the `ui` pod here, so I would give this a different prefix

File: frontend/ui/BUILD [hunk 2]
```diff
@@ -52,7 +52,7 @@ web_app(
 pkg_tar(
     name = "bundle_tar",
     srcs = [":bundle"],
-    package_dir = "/usr/share/nginx/html/",
```
Comment (chrismatix) [inline frontend/ui/BUILD:55]: It's a bit weird that the ui package has to be aware of this change. I would prefer that we rewrite the path by stripping the prefix [in traefik](https://doc.traefik.io/traefik/middlewares/stripprefix/)

File: docs/docs/getting-started/installation/vagrant.md [hunk 3]
```diff
@@ -59,16 +57,16 @@ If you want to customize your `Airy Core` instance please see our [Configuration
...
 `status` command.
 
 ## Access the UI
 
-The UI can be accessed through http://ui.airy.
+The UI can be accessed through http://airy.core/ui/.
 
 The frontend UI for the Airy chat plugin can be accessed through
-http://chatplugin.airy/example. Refer to [the Airy Live Chat
+http://airy.core/chatplugin//example. Refer to [the Airy Live Chat
```
Comment (lucapette) [inline docs/docs/getting-started/installation/vagrant.md:69]:
```suggestion
http://airy.core/chatplugin/example. Refer to [the Airy Live Chat
```

File: docs/docs/sources/chat-plugin.md [hunk 4]
```diff
@@ -46,16 +46,16 @@ the `<head>` section:
 You must replace `CHANNEL_ID` with the channel id obtained when
 [connecting](#connecting-a-channel) the source and `SCRIPT_HOST` with the host
 of your Chat Plugin server. When using the local vagrant environment
-`SCRIPT_HOST` must be set to `chatplugin.airy`.
+`SCRIPT_HOST` must be set to `airy.core`.
 
 :::note
 
-`chatplugin.airy` is not publicly accessible. The setup will only work for local web pages.
+`airy.core/chatplugin/` is not publicly accessible. The setup will only work for local web pages.
```
Comment (lucapette) [inline docs/docs/sources/chat-plugin.md:53]:
```suggestion
`airy.core/chatplugin` is not publicly accessible. The setup will only work for local web pages.
```

File: docs/docs/sources/chat-plugin.md [hunk 5]
```diff
@@ -46,16 +46,16 @@ the `<head>` section:
...
 
 :::
 
 To test the setup, replace the `CHANNEL_ID` in the URL
-`http://chatplugin.airy/example?channel_id=CHANNEL_ID` and open it in your
+`http://airy.core/chatplugin//example?channel_id=CHANNEL_ID` and open it in your
```
Comment (lucapette) [inline docs/docs/sources/chat-plugin.md:58]:
```suggestion
`http://airy.core/chatplugin/example?channel_id=CHANNEL_ID` and open it in your
```

File: frontend/chat-plugin/BUILD [hunk 6]
```diff
@@ -39,7 +41,7 @@ web_library(
     module_deps = module_deps,
     output = {
         "libraryExport": "AiryWidget",
-        "publicPath": "/",
+        "publicPath": "/chatplugin/",
```
Comment (lucapette) [inline frontend/chat-plugin/BUILD:44]: trailing `/` probably not needed? I'm not sure

File: frontend/ui/src/index.tsx [hunk 7]
```diff
@@ -10,11 +10,12 @@ import './index.scss';
 import './assets/scss/reset.scss';
 import '@airyhq/components/dist/main.css';
 
+console.log("public path", process.env.PUBLIC_PATH);
```
Comment (lucapette) [inline frontend/ui/src/index.tsx:13]: 💣

File: infrastructure/scripts/provision/prerequisites.sh [hunk 8]
```diff
@@ -6,7 +6,7 @@ echo "http://dl-cdn.alpinelinux.org/alpine/edge/community" >> /etc/apk/repositor
 apk update
 apk add --no-cache wget unzip jq yq bash-completion
 
-curl -sfL https://get.k3s.io | sh -
+curl -sfL https://get.k3s.io | INSTALL_K3S_EXEC="--disable=traefik" sh -s -
```
Comment (lucapette) [inline infrastructure/scripts/provision/prerequisites.sh:9]: :)

**Review by lucapette — CHANGES_REQUESTED** [review]
Looking good!

Minor suggestions for the docs. I requested changes so we make sure we remove the console.log statement

File: integration/ui/cypress.json [hunk 9]
```diff
@@ -1,5 +1,5 @@
 {
-    "baseUrl": "http://ui.airy",
+    "baseUrl": "http://airy.core",
```
Comment (lucapette) [inline integration/ui/cypress.json:2]:
```suggestion
    "baseUrl": "http://airy.core/ui",
```

**Pull request merged**